src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -156,13 +173,45 @@ assign(SyntheticEvent.prototype, {
    * `PooledClass` looks for `destructor` on each instance it releases.
    */
   destructor: function() {
+    var setNull = setToNullOrWarning.bind(this);
     var Interface = this.constructor.Interface;
-    for (var propName in Interface) {
-      this[propName] = null;
+    for (var interfacePropName in Interface) {
+      setNull(interfacePropName);
+    }
+    otherNullableProps.forEach(setNull);
+    this._destructored = true;
+
+    function setToNullOrWarning(propName) {
```

Comment:
I can't speak for React maintainers but even a function call seems like more indirection compared to a direct assignment. It might be better to do something like

```js
if (__DEV__) {
  function getPooledWarningPropertyDefinition() {
    // ...
  }
}

// ...

var Interface = this.constructor.Interface;
for (var interfacePropName in Interface) {
  if (__DEV__) {
    Object.defineProperty(this, interfacePropName, getPooledWarningPropertyDefinition(interfacePropName))
  } else {
    this[interfacePropName] = null;
  }
}

if (__DEV__) {
  Object.defineProperty(this, 'nativeEvent', getPooledWarningPropertyDefinition('nativeEvent'));
} else {
  this.nativeEvent = null;
}
```

This way the impact on prod is minimal. 

However this is just my opinion. What I would do is to grep the code for more `warning` calls inside getters, and see how their codepaths differ from production codepaths. It is very likely that this particular pattern (“assign a descriptor proxy in dev and a value in prod”) already exists elsewhere in the codebase, and check how it’s usually done.